### PR TITLE
feat(platform): updated the empty state of dashboard

### LINKED ISSUE #485: PLATFORM: Update the empty state for dashboard based on new Figma design
## Description
You need to update the current empty dashboard state with the current empty state design in Figma.

## Scope of Work
You need to update the part of the codebase with the [current Figma design](https://www.figma.com/design/Du3qt5NWbbnfMgtU5CQYCU/Keyshade-UI?node-id=875-6883&node-type=frame&t=0d00kh02HblhIGe3-0)
https://github.com/keyshade-xyz/keyshade/blob/256d65974babd17d525ce2a41a5e17136e6ac12e/apps/platform/src/app/(main)/page.tsx#L176-L178

### PR DESCRIPTION
**Pull request by kriptonian1**
## Description

Updated the empty state of the dashboard

Fixes #485

## Mentions

@rajdip-b @poswalsameer 

## Screenshots of relevant screens

image

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [x] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

___

### **PR Type**
Enhancement

___

### **Description**
- Introduced a new state `isProjectEmpty` to determine if the project list is empty and update the UI accordingly.
- Enhanced the empty state of the dashboard by integrating the `FolderSVG` and a new message encouraging users to start their first project.
- Added a button in the empty state to facilitate the creation of a new project, improving user experience.
- Updated the dashboard to conditionally display project-related elements only when projects exist.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Add FolderSVG to Dashboard SVG Exports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/public/svg/dashboard/index.ts

<li>Added import for <code>FolderSVG</code>.<br> <li> Exported <code>FolderSVG</code> for use in other components.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/522/files#diff-1dfb1c99f5af081b51104dd0547394e429b28ece84a14b45974f58dfac5d08e8">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Update Dashboard Empty State with New Design</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/app/(main)/page.tsx

<li>Added state <code>isProjectEmpty</code> to track if there are no projects.<br> <li> Updated UI to conditionally render elements based on <code>isProjectEmpty</code>.<br> <li> Integrated <code>FolderSVG</code> in the empty state UI.<br> <li> Added a button to create a project in the empty state.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/522/files#diff-d6c7ffe7a9ea578ea4b0a5587f2759a7a1eac9a7c7d6dfae25ddd0bae48ea8e2">+23/-10</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Review by rajdip-b — APPROVED** [review]

**Review by poswalsameer — CHANGES_REQUESTED** [review]

File: apps/platform/src/app/(main)/page.tsx [hunk 1]
```diff
@@ -30,6 +31,7 @@ import { Projects } from '@/lib/api-functions/projects'
 
 export default function Index(): JSX.Element {
   const [isSheetOpen, setIsSheetOpen] = useState<boolean>(false)
+  const [isProjectEmpty, setIsProjectEmpty] = useState(true)
```
[the remark this answers is not in the document]
Comment (kriptonian1) [reply apps/platform/src/app/(main)/page.tsx:34]: I want to conditionally render the header and the add button, so thought to organise it this way. 
Open for discussion, Should we do it like this or by directly checking the size of the array.

[the remark this answers is not in the document]
Comment (rajdip-b) [reply apps/platform/src/app/(main)/page.tsx:34]: I think using the array size would be better. We should release memory wherever we can.

[the remark this answers is not in the document]
Comment (poswalsameer) [reply apps/platform/src/app/(main)/page.tsx:34]: > I want to conditionally render the header and the add button, so thought to organise it this way. Open for discussion, Should we do it like this or by directly checking the size of the array. @poswalsameer @rajdip-b

Got your point, but the conditional rendering can also be done through the size of the `projects` array, so I think this state can be ignored for now. Although, this won't be a big optimization but still we can go with this if we want to.

[the remark this answers is not in the document]
Comment (rajdip-b) [reply apps/platform/src/app/(main)/page.tsx:34]: @kriptonian1 ig you shoud revert to using array size and we can close this

[the remark this answers is not in the document]
Comment (kriptonian1) [reply apps/platform/src/app/(main)/page.tsx:34]: okey will do that

**Pull request merged**